File: mili-server/src/main/java/fun/bm/mili/config/modules/experiment/CommandConfig.java
package fun.bm.mili.config.modules.experiment;

import me.earthme.luminol.config.IConfigModule;
import me.earthme.luminol.config.flags.ConfigClassInfo;
import me.earthme.luminol.config.flags.ConfigInfo;
import me.earthme.luminol.config.flags.TransformedConfig;
import me.earthme.luminol.enums.EnumConfigCategory;

@ConfigClassInfo(category = EnumConfigCategory.EXPERIMENT, name = "command")
public class CommandConfig implements IConfigModule {
    @ConfigInfo(name = "tick_command_enabled", comments =
            """
                    允许使用 tick 命令""")
    public static boolean tick = false;

    @ConfigInfo(name = "function_command_enabled", comments =
            """
                    允许使用 function 命令""")
    public static boolean function = false;

    @TransformedConfig(name = "enable-waypoint", directory = {"experiment", "waypoint bar"})
    @TransformedConfig(name = "enable-waypoint", directory = {"experiment", "waypoint_bar"})
    @ConfigInfo(name = "waypoint_command_enabled", comments =
            """
                    允许使用 waypoint 命令和定位栏""")
    public static boolean waypoint = false;

    @ConfigInfo(name = "scoreboard_command_enabled", comments =
            """
                    允许使用 scoreboard 命令""")
    public static boolean scoreboard = false;

    @ConfigInfo(name = "enabled", directory = {"save_all_command"}, comments =
            """
                    允许使用 save-all 命令""")
    public static boolean saveAll = false;

    @ConfigInfo(name = "log_all_process", directory = {"save_all_command"}, comments =
            """
                    将 save-all 命令的所有过程记录到控制台""")
    public static boolean logAllProcess = false;

    @ConfigInfo(name = "save_all_command_timeout", directory = {"save_all_command"}, comments = """
            区块保存超时报告前的最大秒数。""")
    public static long saveAllTimeout = 30;
}

File: mili-server/src/main/kotlin/fun/bm/mili/config/modules/experiment/CommandConfig.kt
package `fun`.bm.mili.config.modules.experiment

import com.electronwill.nightconfig.core.file.CommentedFileConfig
import `fun`.bm.mili.config.modules.ConfigModule
import me.earthme.luminol.config.flags.ConfigClassInfo
import me.earthme.luminol.config.flags.ConfigInfo
import me.earthme.luminol.enums.EnumConfigCategory

@ConfigClassInfo(category = EnumConfigCategory.EXPERIMENT, name = "command")
class CommandConfig : ConfigModule {
    companion object {
        @ConfigInfo(name = "function", comments = "Enable /function command") @JvmField var function = true
        @ConfigInfo(name = "scoreboard", comments = "Enable /scoreboard command") @JvmField var scoreboard = true
        @ConfigInfo(name = "tick", comments = "Enable /tick command") @JvmField var tick = true
        @ConfigInfo(name = "save-all", comments = "Enable /save-all command") @JvmField var saveAll = true
    }
    override fun onLoaded(c: CommentedFileConfig) {}
    override fun onUnloaded(c: CommentedFileConfig) {}
}
